stubs/algopy-stubs/_box.pyi

```diff
@@ -0,0 +1,193 @@
+import typing
+from typing import Generic, TypeVar
+
+from algopy import Bytes, UInt64
+
+_TContent = TypeVar("_TContent")
+_TKey = TypeVar("_TKey")
+
+class Box(Generic[_TContent]):
+    """
+    Box abstracts the reading and writing of a single value to a single statically keyed box.
+    The box size will be reconfigured dynamically to fit the size of the value being assigned to it.
+    """
+
+    def __init__(
+        self,
+        type_: type[_TContent],
+        /,
+        *,
+        key: Bytes | bytes = ...,
+        description: str = ...,
+    ) -> None: ...
+    def __bool__(self) -> bool:
+        """
+        Returns True if the box has a value set, regardless of the truthiness of the contents
+        of the box
+        """
+    @property
+    def value(self) -> _TContent:
+        """Retrieve the contents of the box. Fails if the box has not been created."""
+    @value.setter
+    def value(self, value: _TContent) -> None:
+        """Write _value_ to the box. Creates the box if it does not exist."""
+    @value.deleter
+    def value(self) -> None:
+        """Delete the box"""
+    def get(self, *, default: _TContent) -> _TContent:
+        """
+        Retrieve the contents of the box, or return the default value if the box has not been
+        created.
+        @param default: The default value to return if the box has not been created
+        """
+    def maybe(self) -> tuple[_TContent, bool]:
+        """
+        Retrieve the contents of the box if it exists, and return a boolean indicating if the box
+        exists.
+
+        """
+    @property
+    def length(self) -> UInt64:
+        """
+        Get the length of this Box. Fails if the box does not exist
+        """
+
+class BoxBlob:
+    """
+    BoxBlob abstracts the reading and writing of boxes containing raw binary data. The size is
+    configured manually, and can be set to values larger than what the AVM can handle in a single
+    value.
+    """
+
+    def __init__(
+        self,
+        /,
+        *,
+        key: Bytes | bytes = ...,
+        description: str = ...,
+    ) -> None: ...
+    def __bool__(self) -> bool:
+        """Returns True if the box has a value set, regardless of the truthiness of that value"""
+    @typing.overload
+    def create(self, value: Bytes | bytes) -> None:
+        """
+        Creates a box with the specified value. Fails if the box already exists with a
+        different size to `len(value)`
+        """
+    @typing.overload
+    def create(self, size: UInt64 | int) -> bool:
```

Comment:
Should this be it's own method rather than an overload? Since in the bytes overload it stores the passed value, but in this uint64 overload it is just allocating the space